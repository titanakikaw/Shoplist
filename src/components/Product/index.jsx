import { connect } from "react-redux";
// import { addCart, deleteCart, updateCart } from "../../actions/cartActions";
import Product from "./Product";

const mapStateToProps = ({ cart, loading }, { product }) => ({
  cartItem: cart.find((x) => x.productId === product.id),
  isLoading: loading.some((x) => x.loadingId === product.id),
});

const mapDispatchToProps = (dispatch) => ({
  addToCart: (data) =>
    dispatch({
      type: "ADD_CART_REQUEST",
      payload: {
        url: "660/cart",
        method: "post",
        data,
      },
      meta: {
        loadingId: data.productId,
      },
    }),
  updateCartItem: (data) =>
    dispatch({
      type: "UPDATE_CART_REQUEST",
      payload: {
        url: `660/cart/${data.id}`,
        method: "put",
        data,
      },
      meta: {
        loadingId: data.productId,
      },
    }),
  deleteCartItem: (data) =>
    dispatch({
      type: "DELETE_CART_REQUEST",
      payload: {
        url: `660/cart/${data.id}`,
        method: "delete",
        data,
      },
      meta: {
        loadingId: data.productId,
      },
    }),
});

export default connect(mapStateToProps, mapDispatchToProps)(Product);
