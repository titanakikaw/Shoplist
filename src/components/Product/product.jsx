import { StarIcon } from "@heroicons/react/24/outline";
import clsx from "clsx";
import React from "react";

const Product = ({
  product,
  cartItem,
  isLoading,
  addToCart,
  updateCartItem,
  deleteCartItem,
}) => {
  return (
    <div className="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8 grid w-full grid-cols-1 items-start gap-y-8 gap-x-6 sm:grid-cols-12 lg:gap-x-8 my-10">
      <div className="aspect-w-2 aspect-h-3 overflow-hidden rounded-lg bg-gray-100 sm:col-span-2 lg:col-span-3">
        <img
          src={product.image}
          alt={product.title}
          className="object-cover object-center"
        />
      </div>
      <div className="sm:col-span-10 lg:col-span-9 grid h-full">
        <h2 className="text-2xl font-bold text-gray-900 sm:pr-12">
          {product.title}
        </h2>

        <section aria-labelledby="information-heading" className="mt-2">
          <h3 id="information-heading">{product.description}</h3>

          <p className="text-2xl text-gray-900">{product.price}</p>

          {/* Reviews */}
          <div className="mt-6">
            <h4 className="sr-only">Reviews</h4>
            <div className="flex items-center">
              <div className="flex items-center">
                {[0, 1, 2, 3, 4].map((rating) => (
                  <StarIcon
                    key={rating}
                    className={clsx("h-5 w-5 flex-shrink-0 text-gray-900", {
                      "text-gray-200": product.rating.rate <= rating,
                    })}
                    aria-hidden="true"
                  />
                ))}
              </div>
              <p className="sr-only">{product.rating.rate} out of 5 stars</p>
              <a
                href="/"
                className="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
              >
                {product.rating.count} reviews
              </a>
            </div>
          </div>
        </section>

        <section aria-labelledby="options-heading" className="mt-4">
          {cartItem ? (
            <div className="flex items-center">
              <button
                type="submit"
                disabled={isLoading}
                className="flex flex-1 w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 py-3 px-8 text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 disabled:bg-slate-500 disabled:cursor-wait"
                onClick={() =>
                  updateCartItem({
                    ...cartItem,
                    quantity: cartItem.quantity + 1,
                  })
                }
              >
                +
              </button>
              <p className="flex-1 text-2xl font-bold text-center">
                {cartItem.quantity}
              </p>
              <button
                type="submit"
                disabled={isLoading}
                className="flex flex-1 w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 py-3 px-8 text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2  disabled:bg-slate-500 disabled:cursor-wait"
                onClick={() => {
                  if (cartItem.quantity <= 1) {
                    deleteCartItem(cartItem);
                  } else {
                    updateCartItem({
                      ...cartItem,
                      quantity: cartItem.quantity - 1,
                    });
                  }
                }}
              >
                -
              </button>
            </div>
          ) : (
            <button
              type="submit"
              className="flex w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 py-3 px-8 text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
              onClick={() =>
                addToCart({
                  productId: product.id,
                  quantity: 1,
                })
              }
            >
              Add to bag
            </button>
          )}
        </section>
      </div>
    </div>
  );
};

export default Product;
