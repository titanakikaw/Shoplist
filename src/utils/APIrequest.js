import axiosInstance from "./axiosInstance";

export default async ({ dispatch, apiData, type, meta }) => {
  try {
    dispatch({
      type: `${type}_REQUEST`,
      meta,
    });
    const res = await axiosInstance(apiData);
    dispatch({
      type: `${type}_SUCCESS`,
      payload: apiData.method === "delete" ? apiData.data : res,
      meta,
    });
  } catch (err) {
    dispatch({
      type: `${type}_FAIL`,
      payload: {
        message: err.message,
        title: `${type
          .split("_")
          .map((x, i) => {
            if (i === 0) {
              return `${x[0].toUpperCase()}${x.slice(1).toLocaleLowerCase()}`;
            }
            return x.toLocaleLowerCase();
          })
          .join(" ")} fail`,
      },
      meta,
    });
  }
};
