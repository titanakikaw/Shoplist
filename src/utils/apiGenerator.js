import { call, put } from "redux-saga/effects";
import axiosInstance from "./axiosInstance";

function* apiGenerator({ type, payload, meta }) {
  const match = /(.*)_(REQUEST)/.exec(type);
  const [, actionName] = match;
  try {
    const res = yield call(axiosInstance, payload);
    yield put({
      type: `${actionName}_SUCCESS`,
      payload: actionName === "DELETE_CART" ? payload.data : res,
      meta,
    });
  } catch (error) {
    yield put({ actionName: `${actionName}_FAIL`, payload: error, meta });
  }
}

export default apiGenerator;
