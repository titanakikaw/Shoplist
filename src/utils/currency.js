export const currency = (price) => {
  return new Intl.NumberFormat("en-IN", {
    style: "currency",
    currency: "INR",
  }).format(price);
};
