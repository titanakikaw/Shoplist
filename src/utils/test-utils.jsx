import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configStore from "../configStore";

// As a basic setup, import your same slice reducers

export default function renderWithProviders(ui) {
  const store = configStore();
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper }) };
}
