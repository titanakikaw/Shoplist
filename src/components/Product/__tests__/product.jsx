import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { currency } from "../../../utils/currency";
import Product from "../Product";

const product = {
  id: 1,
  title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
  price: 109.95,
  description:
    "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
  category: "men's clothing",
  image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
  rating: {
    rate: 3.9,
    count: 120,
  },
};

const cartItem = {
  productId: 1,
  quantity: 2,
  id: 1,
};

const addToCart = jest.fn();

const updateCartItem = jest.fn();

const deleteCartItem = jest.fn();

const renderProduct = ({ ...data }) => {
  const props = {
    product,
    isLoading: false,
    addToCart,
    updateCartItem,
    deleteCartItem,
    ...data,
  };
  return render(<Product {...props} />);
};

describe("Product Component For Add Cart", () => {
  let container = null;
  beforeEach(() => {
    container = renderProduct({}).container;
  });

  test("should render Product Component", () => {
    expect(container.firstChild).toBeDefined();
  });

  test("should take snapshot for add cart item", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render Add to bag button", () => {
    const btns = screen.queryAllByRole("button");
    expect(btns.length).toBe(1);
    expect(btns[0].innerHTML).toBe("Add to bag");
  });

  test("should render product Title", () => {
    const title = screen.queryByRole("heading", { level: 2 });
    expect(title).not.toBeNull();
    expect(title.innerHTML).toBe(product.title);
  });

  test("should render product desc", () => {
    const desc = screen.queryByRole("heading", { level: 3 });
    expect(desc).not.toBeNull();
    expect(desc.innerHTML).toBe(product.description);
  });

  test("should render price", () => {
    const price = screen.queryByTestId("price");
    expect(price).not.toBeNull();
    expect(price.innerHTML).toBe(currency(product.price));
  });

  test("should click add to bag button", () => {
    const btns = screen.queryAllByRole("button");
    fireEvent.click(btns[0]);
    expect(addToCart).toBeCalledTimes(1);
    expect(addToCart).toBeCalledWith({ productId: product.id, quantity: 1 });
  });
});

describe("Product Component For Update Cart", () => {
  let container = null;
  beforeEach(() => {
    container = renderProduct({ cartItem }).container;
  });

  test("should render Update Product Component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render Add to bag button", () => {
    const btns = screen.queryAllByRole("button");
    expect(btns.length).toBe(2);
    expect(btns[0].innerHTML).toBe("+");
    expect(btns[1].innerHTML).toBe("-");
  });

  test("should update add quantity", () => {
    const btns = screen.queryAllByRole("button");
    expect(btns[0].innerHTML).toBe("+");
    fireEvent.click(btns[0]);
    expect(updateCartItem).toBeCalledTimes(1);
    expect(updateCartItem).toBeCalledWith({
      ...cartItem,
      quantity: cartItem.quantity + 1,
    });
  });

  test("should update minus quantity", () => {
    const btns = screen.queryAllByRole("button");
    expect(btns[1].innerHTML).toBe("-");
    fireEvent.click(btns[1]);
    expect(updateCartItem).toBeCalledTimes(1);
    expect(updateCartItem).toBeCalledWith({
      ...cartItem,
      quantity: cartItem.quantity - 1,
    });
  });
  
  

});
