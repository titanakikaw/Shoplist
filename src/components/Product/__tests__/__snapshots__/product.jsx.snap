// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component For Add Cart should take snapshot for add cart item 1`] = `
<div
  class="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8 grid w-full grid-cols-1 items-start gap-y-8 gap-x-6 sm:grid-cols-12 lg:gap-x-8 my-10"
>
  <div
    class="aspect-w-2 aspect-h-3 overflow-hidden rounded-lg bg-gray-100 sm:col-span-2 lg:col-span-3"
  >
    <img
      alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
      class="object-cover object-center"
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
  </div>
  <div
    class="sm:col-span-10 lg:col-span-9 grid h-full"
  >
    <h2
      class="text-2xl font-bold text-gray-900 sm:pr-12"
    >
      Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
    </h2>
    <section
      aria-labelledby="information-heading"
      class="mt-2"
    >
      <h3
        id="information-heading"
      >
        Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
      </h3>
      <p
        class="text-2xl text-gray-900"
        data-testid="price"
      >
        ₹109.95
      </p>
      <div
        class="mt-6"
      >
        <h4
          class="sr-only"
        >
          Reviews
        </h4>
        <div
          class="flex items-center"
        >
          <div
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900 text-gray-200"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <p
            class="sr-only"
          >
            3.9
             out of 5 stars
          </p>
          <a
            class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
            href="/"
          >
            120
             reviews
          </a>
        </div>
      </div>
    </section>
    <section
      aria-labelledby="options-heading"
      class="mt-4"
    >
      <button
        class="flex w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 py-3 px-8 text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
        type="submit"
      >
        Add to bag
      </button>
    </section>
  </div>
</div>
`;

exports[`Product Component For Update Cart should render Update Product Component 1`] = `
<div
  class="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8 grid w-full grid-cols-1 items-start gap-y-8 gap-x-6 sm:grid-cols-12 lg:gap-x-8 my-10"
>
  <div
    class="aspect-w-2 aspect-h-3 overflow-hidden rounded-lg bg-gray-100 sm:col-span-2 lg:col-span-3"
  >
    <img
      alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
      class="object-cover object-center"
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
  </div>
  <div
    class="sm:col-span-10 lg:col-span-9 grid h-full"
  >
    <h2
      class="text-2xl font-bold text-gray-900 sm:pr-12"
    >
      Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
    </h2>
    <section
      aria-labelledby="information-heading"
      class="mt-2"
    >
      <h3
        id="information-heading"
      >
        Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
      </h3>
      <p
        class="text-2xl text-gray-900"
        data-testid="price"
      >
        ₹109.95
      </p>
      <div
        class="mt-6"
      >
        <h4
          class="sr-only"
        >
          Reviews
        </h4>
        <div
          class="flex items-center"
        >
          <div
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-900 text-gray-200"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <p
            class="sr-only"
          >
            3.9
             out of 5 stars
          </p>
          <a
            class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
            href="/"
          >
            120
             reviews
          </a>
        </div>
      </div>
    </section>
    <section
      aria-labelledby="options-heading"
      class="mt-4"
    >
      <div
        class="flex items-center"
      >
        <button
          class="flex flex-1 w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 py-3 px-8 text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 disabled:bg-slate-500 disabled:cursor-wait"
          type="submit"
        >
          +
        </button>
        <p
          class="flex-1 text-2xl font-bold text-center"
        >
          2
        </p>
        <button
          class="flex flex-1 w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 py-3 px-8 text-base font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2  disabled:bg-slate-500 disabled:cursor-wait"
          type="submit"
        >
          -
        </button>
      </div>
    </section>
  </div>
</div>
`;
